api/types/mount/mount.go

```diff
@@ -80,6 +80,7 @@ const (
 // BindOptions defines options specific to mounts of type "bind".
 type BindOptions struct {
 	Propagation Propagation `json:",omitempty"`
+	NoRecursive bool        `json:",omitempty"`
```

Comment:
Does `omitempty` actually work for a `bool` (not a pointer)? If not; perhaps we should make it a pointer so that it can be kept empty (and we can distinguish `false` from `not set`)

We should also make sure that this option is ignored on older API versions; i.e., if a request is sent to API version < 1.40, the `NoRecursive` option should be set to `nil` (or `false`); similar to https://github.com/moby/moby/blob/8e610b2b55bfd1bfa9436ab110d311f5e8a74dcb/api/server/router/container/container_routes.go#L463-L466


Same for `createService`; https://github.com/moby/moby/blob/9f296d1e6fccd9f73bae345e4aad4f3f6e92fdeb/api/server/router/swarm/cluster_routes.go#L189-L195

And for `updateService`;

https://github.com/moby/moby/blob/9f296d1e6fccd9f73bae345e4aad4f3f6e92fdeb/api/server/router/swarm/cluster_routes.go#L232-L238

Reply:
https://play.golang.org/p/gEZoDu7LdG

> Does omitempty actually work for a bool (not a pointer)? 

Seems yes